### PR TITLE
Allow more audio to be pulled out, plus more

### LINKED ISSUE #2: Revert PR #1
I am so sorry, Matt.

I was working with LIME again today (someone asked me to replace the command-line parameter with a dialog box, which I did), and I realized that I submitted a _completely_ broken [PR](1). I had gotten my Python versions mixed up, and testing LIME on Python 2 instead of 3, like I had thought I was doing. I actually broke the script in the PR: the import is wrong, and it throws an assert error on Python 3.

Since I feel like I need to correct this issue I created, here is a fixed version. It is Python 2 only, just like before, but I've added the message changes I made earlier. I can't do another PR, for GitHub wants to merge other files (like the aforementioned file dialog).

Please accept this commit, and I am so, so sorry about this. I test my code so many times before I commit it, let alone before submitting a PR, so I feel really bad about this.
- [Fixed script](https://github.com/le717/LIME/blob/f92c6e1772609ab5c61d362259e9750ec9541181/lime.py)

### PR DESCRIPTION
**Pull request by le717**
Sorry for the late reply. After I saw Issue #3 and used it, I found that the was more than one error stopping every last piece (or so I guess) of audio from being pulled out. This PR fixes that, so all Exceptions are caught. In addition, this PR also contains....
- Fixes all PEP 8 errors,
- Removes broken Py3K code (I could not make it compatible at all. I'm not knowledgeable enough in Python to do that yet),
- Opens a Tkinter File Dialog if the command-line argument is not invoked (from another version of LIME we talked about in Issue #2)

I am very interesting in this project, as well as a whole international community, because LEGO Island is one of the most beloved LEGO video games, and as a modding community, LIME is of special interest because even the most skilled modders have been unable to open the SI archives to get the files out; LIME is the first successful program to ever do that.

The script has been tested successfully tested on Python 2.7.5 on Windows. :smile:

Oh, and a tip for the `rewrite` branch: when trying to extract the files, don't attempt to convert them or anything. As modders, our job is to get the files out in their original form, document the format and recreate it, and have the game load the new files. :wink:

Oh, and this PR would close #2 if you choose to merge it. :wink:

### CODE REVIEW

File: lime.py [hunk 1]
```diff
@@ -97,73 +99,129 @@ def dumper(self, file):
...
-        #print 'WAV "%s" detected!' % name
-        
-        #out = open("%s.bin" % name, 'wb')
-        #out.write(blob)
-        #out.close()
-        
+
         blob = StringIO(blob[listbegin:])
         audiolist = Chunk(blob, attacher)
+<<<<<<< HEAD
```
Comment (mathom) [inline lime.py]: Looks like you have some leftover merge tags in here

**Comment by mathom** [comment]
Also with the converting: the rewrite branch can already dump binaries if you want to check that out. The WAV conversion is about as simple as it gets because the source format is basically WAV already. It just has some meta data in the original file that doesn't make it over. The original packages are basically [RIFF](https://en.wikipedia.org/wiki/Resource_Interchange_File_Format) (which WAV implements) if you want to read up on how it's stored.

**Comment by mathom** [comment]
Oh, and thanks for the PR!

**Comment by le717** [comment]
Fixed the HEAD error, no idea how that got in there. I also quickly added the ability to extract the audio to a subfolder (in the same directory as LIME) using the name of the archive as the destination. It can get really annoying when everything is extracted to the same location as the script. It works with both relative and absolute paths, in both the command-line and Tkinter file dialog. I also did a bit of clean on my own code from my last PR. Again, successfully tested on Python 2.7.5 on Windows. If you want me to change something, let me know.

**Comment by ghost** [comment]
Hey, it's Nick here, and I just tested your code -- works great! I'm gonna see into merging it now.

**Comment by le717** [comment]
Thanks for adding my changes. I know this will help out a lot in our use. Perhaps when the rewrite is near completion they can be added to that.

> Also with the converting: the rewrite branch can already dump binaries if you want to check that out. The WAV conversion is about as simple as it gets because the source format is basically WAV already. It just has some meta data in the original file that doesn't make it over. The original packages are basically RIFF (which WAV implements) if you want to read up on how it's stored.

I haven't enough skill Python to work with binary and archives (I'm self-taught, only been coding for the first time since January this year), so I can't accurately talk about the archives. I do remember others saying they are heavily RIFF archives, but that is about it. Like I said, if you want, you can always [drop by](http://rockraidersunited.org) and enlighten us on what you know so far.

What exactly do you mean by "dump binaries"? It looks to me as if it displays the entire hex code to each file in the console with no formatting (if that is possible in hex). If that's true, then all the code would need is to be aligned to all the proper offsets (if it's not already), get the filelist of each file in the archive (if there is one) then take the data for each file and write it using the name in the list.

**Pull request merged**